### ISSUE TITLE
Unexpected error in explicit-length-check with boolean logical OR

### ISSUE THREAD

**Issue by stonegray** [sindresorhus/eslint-plugin-unicorn#1000]
When linting code that contains a `.length` on one side of a boolean literal OR, unicorn unexpectedly reports an error from the `explicit-length-check` rule. 

This occurs when using the default/recommended rule set.

**Example input and configuration**
```js
const os = require('os');

const options = {};

const example = +options.cpus || os.cpus().length
```
```json eslintrc
{
	"plugins": [
		"unicorn"
	],
	"parserOptions": {
		"ecmaVersion": 2020
	},
	"rules": {
		"unicorn/explicit-length-check": "error"
	}
}
```

Expected output: eslint shows no error

Observed output:
```text
/private/tmp/test/test.js
  5:33  error  Use `.length > 0` when checking length is not zero  unicorn/explicit-length-check

✖ 1 problem (1 error, 0 warnings)
```

**Comment by sindresorhus** [comment]
Similar to https://github.com/sindresorhus/eslint-plugin-unicorn/issues/980.

**Comment by fregante** [comment]
Duplicate of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1840
